File: internal/models/model_config.go
package models

import (
	"time"
)

// ModelProvider 模型提供商
type ModelProvider struct {
	ID          string    `db:"id" json:"id"`
	TenantID    string    `db:"tenant_id" json:"tenant_id"`
	Name        string    `db:"name" json:"name"`
	DisplayName string    `db:"display_name" json:"display_name"`
	BaseURL     string    `db:"base_url" json:"base_url"`
	APIKey      string    `db:"api_key" json:"api_key"`
	Enabled     bool      `db:"enabled" json:"enabled"`
	CreatedAt   time.Time `db:"created_at" json:"created_at"`
	UpdatedAt   time.Time `db:"updated_at" json:"updated_at"`
}

// ModelConfig 模型配置
type ModelConfig struct {
	ID          string  `db:"id" json:"id"`
	TenantID    string  `db:"tenant_id" json:"tenant_id"`
	ProviderID  string  `db:"provider_id" json:"provider_id"`
	ModelName   string  `db:"model_name" json:"model_name"`
	DisplayName string  `db:"display_name" json:"display_name"`
	Temperature float64 `db:"temperature" json:"temperature"`
	MaxTokens   int     `db:"max_tokens" json:"max_tokens"`
	IsDefault   bool    `db:"is_default" json:"is_default"`
	Enabled     bool    `db:"enabled" json:"enabled"`
	CreatedAt   time.Time `db:"created_at" json:"created_at"`
	UpdatedAt   time.Time `db:"updated_at" json:"updated_at"`
}

// ModelProviderWithConfigs 带配置的提供商
type ModelProviderWithConfigs struct {
	ModelProvider
	Configs []ModelConfig `json:"configs"`
}

File: frontend/src/api/modelConfig.ts
import client from './client';

export interface ModelProvider {
  id: string;
  tenant_id: string;
  name: string;
  display_name?: string;
  type: string;
  base_url: string;
  api_key?: string;
  enabled?: boolean;
  created_at: string;
  updated_at: string;
}

export interface ModelConfig {
  id: string;
  tenant_id: string;
  provider_id: string;
  model_name: string;
  display_name?: string;
  temperature?: number;
  max_tokens?: number;
  is_default: boolean;
  enabled?: boolean;
  created_at: string;
  updated_at: string;
}

export interface ValidateModelRequest {
  base_url: string;
  api_key: string;
  model: string;
  call_type?: string;
  response_type?: string;
}

export interface ValidateModelResponse {
  success: boolean;
  message: string;
  api_type?: string;
  response_type?: string;
  supports_stream?: boolean;
  supports_tools?: boolean;
  token_field_type?: string;
  prompt_tokens?: number;
  total_tokens?: number;
}

export const modelConfigApi = {
  // Providers
  listProviders: (tenantId: string) =>
    client.get<ModelProvider[]>(`/tenants/${tenantId}/model-providers`),
  getProvider: (tenantId: string, id: string) =>
    client.get<ModelProvider>(`/tenants/${tenantId}/model-providers/${id}`),
  createProvider: (tenantId: string, data: Partial<ModelProvider>) =>
    client.post<ModelProvider>(`/tenants/${tenantId}/model-providers`, data),
  updateProvider: (tenantId: string, id: string, data: Partial<ModelProvider>) =>
    client.put<ModelProvider>(`/tenants/${tenantId}/model-providers/${id}`, data),
  deleteProvider: (tenantId: string, id: string) =>
    client.delete(`/tenants/${tenantId}/model-providers/${id}`),

  // Configs
  listConfigs: (tenantId: string) =>
    client.get<ModelConfig[]>(`/tenants/${tenantId}/model-configs`),
  getConfig: (tenantId: string, id: string) =>
    client.get<ModelConfig>(`/tenants/${tenantId}/model-configs/${id}`),
  getDefaultConfig: (tenantId: string) =>
    client.get<ModelConfig>(`/tenants/${tenantId}/model-configs/default`),
  createConfig: (tenantId: string, data: Partial<ModelConfig>) =>
    client.post<ModelConfig>(`/tenants/${tenantId}/model-configs`, data),
  updateConfig: (tenantId: string, id: string, data: Partial<ModelConfig>) =>
    client.put<ModelConfig>(`/tenants/${tenantId}/model-configs/${id}`, data),
  deleteConfig: (tenantId: string, id: string) =>
    client.delete(`/tenants/${tenantId}/model-configs/${id}`),
  setDefault: (tenantId: string, id: string) =>
    client.put(`/tenants/${tenantId}/model-configs/${id}/default`),
  validateModel: (tenantId: string, data: ValidateModelRequest) =>
    client.post<ValidateModelResponse>(`/tenants/${tenantId}/model-configs/validate`, data),
};
